docs/parameterized-function-expressions.md

```diff
@@ -0,0 +1,179 @@
+
+```lua
+-- 1999 is a number, so moneyBinding is React.Binding<T>
+local moneyBinding = React.createBinding(1999)
+```
+
+In other cases, the type is more complex. For example, if we want money to be `number?`, then we cannot simply pass in `nil`, as that will infer as `React.Binding<nil>`. Furthermore, while there are theoretically cases like these where Luau can expand out a `nil` into `number?`, it is still beneficial to want this to be restricted at the type to prevent confusing debugging or potential for the type to infer wider than intended.
+
+There are two ways to do this. One is to declare the type of `moneyBinding` itself to `React.Binding`. This would look like...
+
+```lua
+local moneyBinding: React.Binding<number?> = React.createBinding(nil)
+```
+
+This is not only quite verbose, but requires that the user know the exact type of the return value of `React.createBinding`, *and* requires that it really be as trivial as just passing in the types. For example, `React.Binding` could be an even more complex type that asks for some sort of initial state type, or setter type, and so on. In other cases, it might even be defined as an inline table, or as a non-exported type.
+
+The other alternative is something like:
+
+```lua
+local moneyBinding = React.createBinding(nil :: number?)
+```
+
+However, `::` has the ability to cast things in unsound manners. Both these calls are accepted with `::` when we would not want it in this case:
+
+```lua
+local function f(x: number?)
+    React.createBinding(x :: number) -- Allowed
+end
+
+type Object = { a: number }
+React.createBinding({} :: Object) -- Allowed
+```
+
+So while it would work in some cases, it provides an extra bit of uncertainty to the correctness of the call.
+
+In other cases, type parameters are used in places where there is no immediate value binding to that type. For example, a Signal type might be defined as:
+
+```lua
+type Signal<Args...> = {
+    fire: (Args...) -> (),
+    connect: ((Args...) -> ()) -> (),
+}
+
+local function createSignal<Args...>(): Signal<Args...>
+```
+
+The usual local assignment bindings work fine here:
+
+```lua
+local playerTookDamageSignal: Signal<Player, number> = createSignal()
+```
+
+...but only if this binding is actually declared through a local. In a case like:
+
+```lua
+Signals.playerTookDamageSignal = createSignal()
+```
+
+...we either have to put this out into another variable above, which increases surface area, or use `::`, which has the problems described previously. This is not a problem specific to function calls, but is one of the places where they would be useful.
+
+## Design
+
+We propose allowing the following:
+
+```lua
+-- Parameterized function call as statement
+f<<T, U>>()
+
+-- Parameterized function call as expression
+local a = f<<T, U>>()
+
+-- Calling through index
+local b = t.f<<T>>()
+local c = t:f<<T>>()
+local d = t["f"]<<T>>()
+
+-- Saving `f<<T>>` as a variable to called later
+local e = f<<T>>
+```

```

Comment:
Might want to lead with this, or something else similar to be clear that the proposal is allowing the expression `prefixexp '<<' TypeList '>>'` as its own expression, and not just  function calls having an optional new piece of syntax.